### PR TITLE
VerifyChecksum() API should preserve options

### PR DESCRIPTION
**Pull request by mikhail-antonov**

### CODE REVIEW

File: db/db_impl.cc [hunk 1]
```diff
@@ -3066,7 +3064,7 @@ Status DBImpl::VerifyChecksum() {
         const auto& fd = vstorage->LevelFilesBrief(i).files[j].fd;
         std::string fname = TableFileName(cfd->ioptions()->cf_paths,
                                           fd.GetNumber(), fd.GetPathId());
-        s = rocksdb::VerifySstFileChecksum(options, env_options, fname);
+        s = rocksdb::VerifySstFileChecksum(GetOptions(), env_options_, fname);
```
Comment (riversand963) [inline db/db_impl.cc]: `GetOptions()` returns the options built from the default column family. We are iterating different column families here, does this mean we need to call `GetOptions` on each of them?

Comment (mikhail-antonov) [reply db/db_impl.cc]: Yeah i think you're right! wasn't sure of the best way to get the options per CF and didn't really think about it, let me fix it

Comment (riversand963) [reply db/db_impl.cc]: From `DBImpl::GetOptions`'s implementation, I think `Options(BuildDBOptions(immutable_db_options_, mutable_db_options_), cfd->GetLatestCFOptions())` should work.

**Comment by mikhail-antonov** [comment]
@riversand963 does it look good? tested with my application it works, going to land once tests pass?

File: db/db_impl.cc [hunk 2]
```diff
@@ -3060,13 +3058,15 @@ Status DBImpl::VerifyChecksum() {
   for (auto& sv : sv_list) {
     VersionStorageInfo* vstorage = sv->current->storage_info();
     ColumnFamilyData* cfd = sv->current->cfd();
+    Options opts = Options(BuildDBOptions(immutable_db_options_,
```
Comment (yiwu-arbug) [inline db/db_impl.cc]: `mutable_db_options_` need to access while holding db mutex. You can call DB::GetDBOptions() instead.

Comment (mikhail-antonov) [reply db/db_impl.cc]: updated

**Pull request closed without merging**